Sources/Optimizely/OptimizelyError.swift

```diff
@@ -118,13 +122,16 @@ extension OptimizelyError: CustomStringConvertible {
         case .conditionNoMatchingAudience(let id):          message = "Audience (\(id)) is not in datafile."
         case .conditionInvalidFormat(let hint):             message = "Condition has an invalid format (\(hint))"
         case .conditionCannotBeEvaluated(let hint):         message = "Condition cannot be evaluated (\(hint))"
-        case .evaluateAttributeInvalidType(let hint):       message = "Evaluation attribute has an invalid value (\(hint))"
-        case .evaluateAttributeValueOutOfRange(let hint):   message = "Evaluation attribute has a value out of range (\(hint))"
+        case .evaluateAttributeInvalidCondition(let condition): message = "Audience condition (\(condition)) has an unsupported condition value. You may need to upgrade to a newer release of the Optimizely SDK."
+        case .evaluateAttributeInvalidType(let condition, let value, let key):       message = "Audience condition (\(condition)) evaluated to UNKNOWN because a value of type (\(value)) was passed for user attribute (\(key))."
+        case .evaluateAttributeValueOutOfRange(let condition, let key):   message = "Audience condition (\(condition)) evaluated to UNKNOWN because the number value for user attribute (\(key)) is not in the range [-2^53, +2^53]."
         case .evaluateAttributeInvalidFormat(let hint):     message = "Evaluation attribute has an invalid format (\(hint))"
-        case .userAttributeInvalidType(let hint):           message = "UserAttribute has an invalid attribute type (\(hint))"
-        case .userAttributeInvalidMatch(let hint):          message = "UserAttribute has an invalid match type (\(hint))"
-        case .userAttributeInvalidFormat(let hint):         message = "UserAttribute has an invalid format (\(hint))"
-
+        case .userAttributeInvalidType(let condition):      message = "Audience condition (\(condition)) uses an unknown condition type. You may need to upgrade to a newer release of the Optimizely SDK."
+        case .userAttributeInvalidMatch(let condition):     message = "Audience condition (\(condition)) uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK."
+        case .userAttributeNilValue(let condition):         message = "Audience condition (\(condition)) evaluated to UNKNOWN because of null value."
+        case .nilAttributeValue(let condition, let key):    message = "Audience condition (\(condition)) evaluated to UNKNOWN because a null value was passed for user attribute (\(key))."
+        case .userAttributeInvalidName(let condition):      message = "Audience condition (\(condition)) evaluated to UNKNOWN because of invalid attribute name."
```

Comment:
I see we need both. Can we change the order of these two lines so "userAttribute" and "attribute" errors can be grouped?